edge/services/helloworld/CreateService.md

```diff
@@ -0,0 +1,164 @@
+
+  ```bash
+  sudo apt install -y git jq
+  ```
+
+  On **macOS**:
+
+  ```bash
+  brew install git jq
+  ```
+
+## <a id=build-publish-your-hw> Building and Publishing Your Own Hello World Example Edge Service
+
+1. `cd` to the directory in which you want to create your new service and then run this command to create the files for a simple edge service and associated Horizon metadata files:
+
+  ```bash
+  hzn dev service new -s myhelloworld -i "$DOCKER_HUB_ID/myhelloworld"
+  ```
+
+  Notice that some project variables are defined in `horizon/hzn.json` and referenced in other files, for example `horizon/service.definition.json`.
+
+2. Edit `service.sh` and change something simple, for example change "Hello" to "Hey there"
+
+  Note: this service is a shell script for brevity, but you can write your service in any language.
+
+3. Build the service docker image:
+
+  ```bash
+  make
+  ```
+
+4. Test the service by running it the simulated agent environment:
+
+  ```bash
+  hzn dev service start -S
+  ```
+
+5. Check that the container is running:
+
+  ```bash
+  sudo docker ps
+  ```
+
+6. See the environment variables Horizon passes into your service container:
+
+  ```bash
+  sudo docker inspect $(sudo docker ps -q --filter name=myhelloworld) | jq '.[0].Config.Env'
+  ```
+
+7. See the docker container running and look at the output:
+
+  on **Linux**:
+
+  ```bash
+  sudo tail -f /var/log/syslog | grep myhelloworld[[]
+  ```
+
+  on **Mac**:
+
+  ```bash
+  sudo docker logs -f $(sudo docker ps -q --filter name=myhelloworld)
+  ```
+
+8. Stop the service:
+
+  ```bash
+  hzn dev service stop
+  ```
+
+9. Have Horizon push your docker image to your registry and publish your service in the Horizon Exchange and see it there:
+
+  ```bash
+  hzn exchange service publish -f horizon/service.definition.json
+  hzn exchange service list
+  ```
+
+10. Publish your edge node deployment pattern in the Horizon Exchange and see it there:
+
+  ```bash
+  hzn exchange pattern publish -f horizon/pattern.json
+  hzn exchange pattern list
+  ```
+
+11. Register your edge node with Horizon to use your deployment pattern:
+
+  ```bash
+  hzn register -p pattern-myhelloworld-$(hzn architecture)
+  ```
+
+12. The edge device will make an agreement with one of the Horizon agreement bots (this typically takes about 15 seconds). Repeatedly query the agreements of this device until the `agreement_finalized_time` and `agreement_execution_start_time` fields are filled in:
+
+  ```bash
+  hzn agreement list
+  ```
+
+13. Once the agreement is made, list the docker container edge service that has been started as a result:
+
+  ```bash
+  sudo docker ps
+  ```
+
+14. See the myhelloworld service output:
+
+  on **Linux**:
+
+  ```bash
+  sudo tail -f /var/log/syslog | grep myhelloworld[[]
+  ```
+
+  on **Mac**:
+
+  ```bash
+  sudo docker logs -f $(sudo docker ps -q --filter name=myhelloworld)
+  ```
+
+15. Unregister your edge node, stopping the myhelloworld service:
```

Comment:
How about "Unregister your edge node to stop the myhelloworld service?"

Reply:
That changes the technical meaning, because unregister does more than just stop the service. I changed it to "Unregister your edge node (which will also stop the myhelloworld service)"